cql3/statements/select_statement.cc

```diff
@@ -1081,22 +1081,24 @@ lw_shared_ptr<const service::pager::paging_state> indexed_table_select_statement
     auto& last_base_pk = last_pos.partition;
     auto* last_base_ck = last_pos.position.has_key() ? &last_pos.position.key() : nullptr;
 
-    bytes_opt indexed_column_value = restrictions::value_for(*cdef, _used_index_restrictions, options);
+    std::vector<bytes_opt> indexed_column_vector = restrictions::value_for(*cdef, _used_index_restrictions, options);
 
     auto index_pk = [&]() {
         if (_index.metadata().local()) {
             return last_base_pk;
         } else {
-            return partition_key::from_single_value(*_view_schema, *indexed_column_value);
+            return partition_key::from_single_value(*_view_schema, *indexed_column_vector.back());
         }
     }();
 
     std::vector<managed_bytes_view> exploded_index_ck;
-    exploded_index_ck.reserve(_view_schema->clustering_key_size());
+    exploded_index_ck.reserve(_view_schema->clustering_key_size() * indexed_column_vector.size());
```

Comment:
I don't understand this. A few lines back, you use indexed_column_vector.back(), indicating you expect one element. Here you treat it as a vector.

Reply:
I believe that we only need one value to create a partition key, but that's not the case with clustering keys. I've tested this setup and it works correct.